hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateMap.java

```diff
@@ -65,9 +65,9 @@ void addPipeline(Pipeline pipeline) throws IOException {
     Preconditions.checkArgument(
         pipeline.getNodes().size() == pipeline.getReplicationConfig()
             .getRequiredNodes(),
-        String.format("Nodes size=%d, replication factor=%d do not match ",
+        "Nodes size=%s, replication factor=%s do not match ",
```

Comment:
can you help why %d to %s
the return values are integers only right? some reason to do so?

Reply:
`Preconditions` (or rather `Strings#lenientFormat` to which it delegates) only supports `%s` placeholders.